bentoml/cli/bento_service.py

```diff
@@ -0,0 +1,536 @@
+                shell=shell, path=path, append=append
+            )
+            click.echo('%s completion installed in %s' % (shell, path))
+        else:
+            click.echo(
+                "'click_completion' is required for BentoML auto-completion, "
+                "install it with `pip install click_completion`"
+            )
+
+    @bentoml_cli.command(
+        help='Containerizes given Bento into a ready-to-use Docker image.',
+        short_help="Containerizes given Bento into a ready-to-use Docker image",
+    )
+    @click.argument("bento", type=click.STRING)
+    @click.option('-p', '--push', is_flag=True)
+    @click.option(
+        '-t',
+        '--tag',
+        help="Optional image tag. If not specified, Bento will generate one from "
+        "the name of the Bento.",
+        required=False,
+    )
+    @click.option(
+        '-u', '--username', type=click.STRING, required=False,
+    )
+    @click.option(
+        '-p', '--password', type=click.STRING, required=False,
+    )
+    def containerize(bento, push, tag, username, password):
+        """Containerize specified BentoService.
+
+        BENTO is the target BentoService to be containerized, referenced by its name
+        and version in format of name:version. For example: "iris_classifier:v1.2.0"
+
+        `bentoml containerize` command also supports the use of the `latest` tag
+        which will automatically use the last built version of your Bento.
+
+        You can provide a tag for the image built by Bento using the
+        `--docker-image-tag` flag. Additionally, you can provide a `--push` flag,
+        which will push the built image to the Docker repository specified by the
+        image tag.
+
+        You can also prefixing the tag with a hostname for the repository you wish
+        to push to.
+        e.g. `bentoml containerize IrisClassifier:latest --push --tag username/iris`
+        would build a Docker image called `username/iris:latest` and push that to
+        Docker Hub.
+
+        By default, the `containerize` command will use the credentials provided by
+        Docker. You may provide your own through `--username` and `--password`.
+        """
+        bento_service_bundle_path = resolve_bundle_path(
+            bento, pip_installed_bundle_path
+        )
+
+        _echo(f"Found Bento: {bento_service_bundle_path}")
+
+        name, version = get_name_version_from_tag(bento)
+        # use tag name and version where applicable
+        if tag is not None:
+            name, v = get_name_version_from_tag(tag)
+            if v is not None:
+                version = v
+
+        name, version = make_bento_name_docker_compatible(name, version)
+
+        full_tag = f"{name}:{version}"
+        if full_tag != bento:
+            _echo(
+                f'Bento tag was changed to be Docker compatible. \n'
+                f'"{bento}" -> "{full_tag}"',
+                CLI_COLOR_WARNING,
+            )
```

Comment:
Assuming you added some validation for the tag parameter, this part could be rewritten to something like this:
```
bs_metadata = load_bento_service_metadata(bento_service_bundle_path)

if ":" not in tag:
   _echo(f"Tag {tag} does not specify an image version, using BentoService version {bs_metadata.version} as docker image version.")
   version= to_docker_tag_compatible_string(bs_metadata.version)
   tag = f"{tag}:{version}"

_echo("Containerizing BentoService {bs_metadata.name}:{bs_metadata.version}, docker image tag: {tag}")

docker_api = docker.APIClient()
...



```